packages/wallet/primitives/src/network.ts

```diff
@@ -1,630 +1,1229 @@
-export type Network = {
+export enum NetworkType {
+  MAINNET = 'mainnet',
+  TESTNET = 'testnet',
+  LOCAL = 'local',
+}
+
+export type BlockExplorerConfig = {
+  name?: string
+  rootUrl: string
+}
+
+export interface Network {
+  chainId: ChainId
+  type: NetworkType
   name: string
+  title?: string
   rpc: string
-  chainId: bigint
-  explorer: string
-  nativeCurrency: {
-    name: string
+  logoURI?: string
+  blockExplorer?: BlockExplorerConfig
+  nativeToken: {
```

Comment:
viem/wagmi chain interface calls it `nativeCurrency` rather than `nativeToken` we might want to have consistency